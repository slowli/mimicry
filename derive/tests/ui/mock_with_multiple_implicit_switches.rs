use mimicry_derive::CallReal;

/// Dummy struct to trick `CallReal` derive logic.
struct RealCallSwitch;

#[derive(CallReal)]
struct MyMock {
    switch: RealCallSwitch,
    another_switch: RealCallSwitch,
}

fn main() {}
