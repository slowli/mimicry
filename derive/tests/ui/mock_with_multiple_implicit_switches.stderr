error: Multiple fields with `RealCallSwitch` type. Mark the expected one with `#[mock(switch)]` attr
 --> tests/ui/mock_with_multiple_implicit_switches.rs:9:5
  |
9 |     another_switch: RealCallSwitch,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
